deploy/sdk/src/dynamo/sdk/core/lib.py

```
@@ -78,11 +117,31 @@ def decorator(inner: Type[G]) -> ServiceInterface[G]:
+        # The type of actual_on_service here would be ServiceInterface[NameOfAbstractClass]
+        # So, T would be NameOfAbstractClass.
+        return provider.create_dependency(on=actual_on_service, **kwargs)
+    elif isinstance(on, ServiceInterface):
+        # This handles both 'on=None' and 'on=SomeServiceInterfaceInstance'
+        # If 'on' is ServiceInterface[K], T could be K. If 'on' is None, T remains unbound here.
+        actual_on_service = on
+        return provider.create_dependency(on=actual_on_service, **kwargs)
+    else:
+        raise TypeError(
+            "depends() expects 'on' to be a ServiceInterface, an AbstractDynamoService type"
+        )
```

Pylint (3.3.7)

[refactor] Unnecessary "elif" after "return", remove the leading "el" from "elif"

(R1705)